### ISSUE TITLE
Section 2: Coordinator - Assign or reassign Applications

### ISSUE THREAD

**Issue by rpradheap** [hackforla/HomeUniteUs#518]
Problem Alignment

The Problem

Coordinators do not currently have a way to look at an overview of all applications in the system that includes

Application that are submitted
Applications that are approved/ Declined
Applications that are ready to be matched
Applications that are matched
Applications that have contract in place
User Story:

As a coordinator, I should be able to assign the application to myself or reassign the applications to another coordinator so other coordinators can help the guest/host

High Level Approach

Dashboard page that shows the status of Guest and Host Application in the system

Solution Alignment

Considerations

Coordinators will be using desktop or tablet--> use responsive design

Acceptance criteria
Success is if Coordinator can successfully

View all applications with status
click the application from the dashboard page to view it
Filter application by Status and Guest/Host.
Able to view invite Guest feature
Key Features

Pictorial - Bar chart representation of the application status of Guest and Host
Placeholder to Invite a Guest to submit an application

Design
https://www.figma.com/file/BNWqZk8SHKbtN1nw8BB7VM/HUU-Everything-Figma-Nov-2022?node-id=2353%3A17352&t=ZM7AykeF7WVNoXkD-0

 Review Existing Design
 Updating Design as Needed
Limitations

Coordinator may not be able to see applications that are yet to be submitted by Guest/Host

**Comment by jasika-ai** [comment]
As a Coordinator, I want to manage my meeting with Host/guest and want to mark “Under Review” during the assessment process.  I want to be able to flag the Guest/Host account for Review/Ban

Acceptance Criteria - 
For each application there is an option to mark the application as “Under Review”. The status for the application should be updated accordingly in the coordinator dashboard as well
Every time the coordinator marks the application as “Under Review” should receive a confirmation prompt to prevent accidental changes.

**Comment by sanya301** [comment]
1. Without assigning we probably don't want them to be able to take action on the application
2. We should also have some sort of resistance/confirmation to re-assigning 
3. Can you rephrase the problem statement

**Comment by lasryariel** [comment]
@sanya301 Can you please confirm which of the following scenarios should be allowed for Coordinator Assignment/Reassignment?

Do we need to consider some type of reassignment approval workflows for some of these scenarios where both coordinators (coordinator currently assigned and the coordinator being reassigned to) have to approve it in some way?

- [ ] Unassigned Host - Assign to Self
- [ ] Unassigned Host - Assign to Other Coordinator
- [ ] Guest/Host Assigned to Self - Reassign to Other Coordinator
- [ ] Guest/Host Assigned to Other Coordinator - Reassign to Self
- [ ] Guest/Host Assigned to Other Coordinator - Reassign to Other Coordinator

**Comment by lasryariel** [comment]
DRAFT - Rewrite Proposal

### Context:

- This story covers a portion of functionality needed for #454 

###  The Problem

Currently Coordinators do not have an online way to manage which Coordinator is assigned to which Guest/Host.

### User Story:

As a Coordinator, I should be able to assign myself to any Guest/Host so that it is clear that I will be assisting that Guest/Host with the application and onboarding process.

### High Level Approach

A field on each Guest/Host which allows for assignment/reassignment of Coordinators.

Note: For Guests the coordinator who sent the invite is assigned as coordinator by default. #520

###  Acceptance criteria
- [ ] Coordinator can assign any Guest/Host to themselves
- [ ] Coordinator can reassign any Guest/Host to another Coordinator
- [ ] Coordinator can easily assign/reassign from the dashboard
- [ ] Coordinator is asked to confirm assignment/reassignment to ensure the action was not performed accidentally

###  Metrics
- Load time when performing assign/reassign action
- Average Number of Coordinators assigned per Guest/Host
- Number of Guest/Hosts with no Coordinator assigned

###  Considerations
- Coordinators will be using desktop or tablet--> use responsive design

###  Design
- https://www.figma.com/file/BNWqZk8SHKbtN1nw8BB7VM/HUU-Everything-Figma-Nov-2022?node-id=2353%3A17352&t=ZM7AykeF7WVNoXkD-0
- [ ] Review existing designs on Figma
- [ ] Edit existing designs as needed
- [ ] Designs are edited and ready for engineering. Link to the Figma: (share Figma link here when ready)

### Engineering
- # 

#### Team Reviews:
- [x] Product - Raji
- [ ] Design - Help Needed
- [ ] Engineering - Help Needed
